### PR TITLE
Fix throwing errors from `Executor.join_async()`

### PR DESCRIPTION
**Pull request by mdemoret-nv**
Exceptions from both python and C++ were getting printed to the log but not raised in python when using `Executor.join_async()`. This boils down to the fact that `future::wait()` was called instead of `future::get()` which actually rethrows the exception.

Added some tests to check this functionality

### CODE REVIEW

File: python/srf/tests/utils.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#include <pybind11/cast.h>
+#include <pybind11/pybind11.h>
+
+#include <stdexcept>
+
+namespace srf::pytests {
+
+namespace py = pybind11;
+
+PYBIND11_MODULE(utils, m)
```
Comment (drobison00) [inline python/srf/tests/utils.cpp:29]: Can we use 'module' instead of m, the short variable names generate style warnings.

**Review by drobison00 — APPROVED** [review]

File: python/srf/tests/utils.cpp (hunk 1 above)
Comment (mdemoret-nv) [reply python/srf/tests/utils.cpp]: Done.

**Review by dagardner-nv — APPROVED** [review]
LGTM

**Comment by mdemoret-nv** [comment]
@gpucibot merge

**Pull request merged**